src/base/search/searchpluginmanager.cpp

```diff
@@ -54,10 +55,30 @@
 
 namespace
 {
-    inline void removePythonScriptIfExists(const QString &scriptPath)
+    void clearPythonCache(const QString &path)
     {
-        Utils::Fs::forceRemove(scriptPath);
-        Utils::Fs::forceRemove(scriptPath + "c");
+        // remove python cache artifacts in `path` and subdirs
+
+        QStringList dirs = {path};
+
+        QDirIterator iter {path, (QDir::AllDirs | QDir::NoDotAndDotDot), QDirIterator::Subdirectories};
+        while (iter.hasNext())
```

Comment:
Don't you mind place all logic inside this loop? This drops two extra loops.

Reply:
I can't find a way to loop through it without using a QStringList.
```c++
~/.local/share/data/qBittorrent/nova3  // passed-in path

~/.local/share/data/qBittorrent/nova3/engines  // first path returned by iter
~/.local/share/data/qBittorrent/nova3/engines/__pycache__
```
And obviously if `QDir::NoDotAndDotDot` is omitted, then paths are visited more than once.

Reply:
```c++
namespace
{
    void removePrecompiledScripts(const QString &path)
    {
        for (const QString file : QDir(path).entryList(QDir::Files)) {
            if (file.endsWith(".pyc"))
                Utils::Fs::forceRemove(file);
        }
    }

    void clearPythonCache(const QString &path)
    {
        // remove python cache artifacts in `path` and subdirs
        removePrecompiledScripts(path);

        QDirIterator iter {path, (QDir::AllDirs | QDir::NoDotAndDotDot), QDirIterator::Subdirectories};
        while (iter.hasNext()) {
            const QString dir = iter.next();
            //  python 2: remove "*.pyc" files
            removePrecompiledScripts(dir);
            // python 3: remove "__pycache__" folders
            if (dir.endsWith("__pycache__"))
                Utils::Fs::removeDirRecursive(dir);
    }
}
```

Reply:
Clearly we prefer different approach (for such shallow dir traversal).
So let me ask this way, does the additional QStringList & loop bothers you much?

Reply:
>Clearly we prefer different approach (for such shallow dir traversal).

What do you mean? Unfortunately it's not so clear for me at this point...

Reply:
>does the additional QStringList & loop bothers you much?

At least, I see no sense in twice traversal through "dirs".

Reply:
At least it's better than your original version.
Although I still don't see any gain in having two loops, neither efficiency nor readability. But I will not insist on my version.